src/sparseml/pytorch/object_detection/train.py

```diff
@@ -427,288 +435,324 @@ def train(hyp, opt, device, callbacks):  # path/to/hyp.yaml or hyp dictionary
+                        "on_model_save", last, epoch, final_epoch, best_fitness, fi
+                    )
+
+                # Stop Single-GPU
+                if RANK == -1 and stopper(epoch=epoch, fitness=fi):
+                    break
+
+                # Stop DDP TODO: known issues
+                # shttps://github.com/ultralytics/yolov5/pull/4576
+                # stop = stopper(epoch=epoch, fitness=fi)
+                # if RANK == 0:
+                #    dist.broadcast_object_list([stop], 0)  # broadcast 'stop'
+                #    to all ranks
+
+            # Stop DPP
+            # with torch_distributed_zero_first(RANK):
+            # if stop:
+            #    break  # must break all DDP ranks
+
+            # end epoch -------------------------------------------------------
+        # end training --------------------------------------------------------
+        if RANK in [-1, 0]:
+            LOGGER.info(
+                f"\n{epochs - start_epoch + 1} epochs completed in "
+                f"{(time.time() - t0) / 3600:.3f} hours."
+            )
+            for f in last, best:
+                if f.exists():
+                    strip_optimizer(f)  # strip optimizers
+                    if f is best:
+                        LOGGER.info(f"\nValidating {f}...")
+                        results, _, _ = val(
+                            data_dict,
+                            batch_size=batch_size // WORLD_SIZE * 2,
+                            imgsz=imgsz,
+                            model=load_checkpoint(
+                                type_="ensemble", weights=best, device=device
+                            )[0],
+                            iou_thres=0.65
+                            if is_coco
+                            else 0.60,  # best pycocotools results at 0.65
+                            single_cls=single_cls,
+                            dataloader=val_loader,
+                            save_dir=save_dir,
+                            save_json=is_coco,
+                            verbose=True,
+                            plots=True,
+                            callbacks=callbacks,
+                            compute_loss=compute_loss,  # val best model with plots
+                            half=half_precision,
                         )
+                        if is_coco:
+                            callbacks.run(
+                                "on_fit_epoch_end",
+                                list(mloss) + list(results) + lr,
+                                epoch,
+                                best_fitness,
+                                fi,
+                            )
+
+            callbacks.run("on_train_end", last, best, plots, epoch, results)
+            LOGGER.info(f"Results saved to {colorstr('bold', save_dir)}")
 
-        callbacks.run("on_train_end", last, best, plots, epoch, results)
-        LOGGER.info(f"Results saved to {colorstr('bold', save_dir)}")
+    else:
```

Comment:
this is a huge chunk of code to indent under `if not one_shot` and the else happens right before we return.  instead, before the train/eval loop, if one_shot is enabled, can we run the block below and then return?

Reply:
Thanks! don't know why I didn't do that before,